### ISSUE TITLE
dex-worker health check reports 500 due to failing oidc connector healthcheck

### ISSUE THREAD

**Issue by colhom** [coreos/dex#222]
The `dex-worker` health check will return 
```
{"status":"error","details":{"code":500,"message":"oidc client provider config empty"}}
```

Even when a valid oidc connector is present. Interesting enough, re-setting the connector via `dexctl` will cause the health check to return 200. Restart dex-worker, and we're back to 500 for the health check.

**Comment by ericchiang** [comment]
I can reproduce this somewhat. On startup there is a brief period when the configuration hasn't been fetched from the remote OIDC provider. During that time dex will return a 500, though it then goes back to 200.

Here's me hitting `/health` every 0.1 seconds.

```
$ ./bin/dex-worker --db-url="postgres://postgres@127.0.0.1:32768/postgres?sslmode=disable"
WARN: Running with no limit on: database idle connections
WARN: Running with no limit on: database open connections
INFO: Loaded IdP connector: id=google type=oidc
INFO: Binding to 127.0.0.1:5556...
INFO: 127.0.0.1 - - [17/Dec/2015:11:20:57 -0800] "GET /health HTTP/1.1" 500 87
INFO: 127.0.0.1 - - [17/Dec/2015:11:20:57 -0800] "GET /health HTTP/1.1" 200 15
INFO: 127.0.0.1 - - [17/Dec/2015:11:20:57 -0800] "GET /health HTTP/1.1" 200 15
INFO: 127.0.0.1 - - [17/Dec/2015:11:20:57 -0800] "GET /health HTTP/1.1" 200 15
INFO: 127.0.0.1 - - [17/Dec/2015:11:20:57 -0800] "GET /health HTTP/1.1" 200 15
INFO: 127.0.0.1 - - [17/Dec/2015:11:20:58 -0800] "GET /health HTTP/1.1" 200 15
INFO: 127.0.0.1 - - [17/Dec/2015:11:20:58 -0800] "GET /health HTTP/1.1" 200 15
```

While this is an issue, this doesn't sound like your problem since it fixes itself pretty quickly.

Are there anything in the dex-worker logs? It sounds like the connector might be having trouble hitting the remote identity provider.

**Comment by ericchiang** [comment]
Closing for now as I believe this might have been fixed by #235 since that PR fixed some client hanging issues. @colhom if you can reproduce this with a more recent version of dex please reopen.
